src/kakarot/precompiles/kakarot_precompiles.cairo

```diff
@@ -82,11 +83,22 @@ namespace KakarotPrecompiles {
             let is_not_deployed = Helpers.is_zero(caller_starknet_address);
 
             if (is_not_deployed != FALSE) {
-                let (revert_reason_len, revert_reason) = Errors.accountNotDeployed();
-                return (
-                    revert_reason_len, revert_reason, CAIRO_PRECOMPILE_GAS, Errors.EXCEPTIONAL_HALT
-                );
+                // Deploy account
+                let (deployed_contract) = Starknet.deploy(caller_address);
+                tempvar caller_starknet_address = deployed_contract;
+                tempvar syscall_ptr = syscall_ptr;
+                tempvar pedersen_ptr = pedersen_ptr;
+                tempvar range_check_ptr = range_check_ptr;
+            } else {
+                tempvar caller_starknet_address = caller_starknet_address;
+                tempvar syscall_ptr = syscall_ptr;
+                tempvar pedersen_ptr = pedersen_ptr;
+                tempvar range_check_ptr = range_check_ptr;
```

Comment:
ok, forget my comment above but here you can do
```suggestion
                // Deploy account
                Starknet.deploy(caller_address);
            } else {
                tempvar syscall_ptr = syscall_ptr;
                tempvar pedersen_ptr = pedersen_ptr;
                tempvar range_check_ptr = range_check_ptr;
                tempvar caller_starknet_address = caller_starknet_address;                
```

Reply:
I don't get it - how do you get the caller starknet address?

Reply:
the return data of the `Starknet.deploy` is in this order

```
syscall_ptr: felt*
pedersen_ptr: HashBuiltin*
range_check_ptr
account_address
```

so you don't need to tempvar, just use the same ordering